torch/jit/annotations.py

```diff
@@ -53,48 +53,73 @@ def __getitem__(self, name):
 
 def get_signature(fn, rcb, loc):
     # Python 3.5 adds support for the nice annotation syntax, so try that first.
+    signature = None
     if PY35:
-        sig = try_real_annotations(fn)
-        if sig is not None:
-            return sig
+        signature = try_real_annotations(fn)
+
+    if signature is None:
+        type_line, source = None, None
+        try:
+            source = dedent(''.join(get_source_lines_and_file(fn)[0]))
+            type_line = get_type_line(source)
+        except TypeError:
+            pass
+        # This might happen both because we failed to get the source of fn, or
+        # because it didn't have any annotations.
+        if type_line is not None:
+            signature = parse_type_line(type_line, rcb, loc)
+
+    return signature
 
-    type_line, source = None, None
-    try:
-        source = dedent(''.join(get_source_lines_and_file(fn)[0]))
-        type_line = get_type_line(source)
-    except TypeError:
-        pass
-    # This might happen both because we failed to get the source of fn, or
-    # because it didn't have any annotations.
-    if type_line is None:
-        return None
 
-    return parse_type_line(type_line, rcb, loc)
+def is_script_callable(the_callable):
+    # A stricter version of `inspect.isroutine` that does not pass for built-in
+    # functions
+    return inspect.isfunction(the_callable) or inspect.ismethod(the_callable)
+
+
+def is_vararg(the_callable):
```

Comment:
What does `is_vararg` mean exactly?  That there is a *args or **kwargs argument?

Reply:
varargs refers only to the `*args` and the terminology comes from `inspect` (it calls `**kwargs` "`varkw`")